base_partner_sequence/models/partner.py

```diff
@@ -0,0 +1,65 @@
+# Copyright 2004-2009 Tiny SPRL (<https://tiny.be>).
+# Copyright 2013 initOS GmbH & Co. KG (<https://www.initos.com>).
+# Copyright 2016 Tecnativa - Vicent Cubells
+# Copyright 2016 Camptocamp - Akim Juillerat (<https://www.camptocamp.com>).
+# License AGPL-3.0 or later (https://www.gnu.org/licenses/agpl.html).
+
+from odoo import _, api, exceptions, models
+
+
+class ResPartner(models.Model):
+    """Assigns 'ref' from a sequence on creation and copying"""
+
+    _inherit = "res.partner"
+
+    def _get_next_ref(self):
+        return self.env["ir.sequence"].next_by_code("res.partner")
+
+    @api.model
+    def create(self, vals):
+        if not vals.get("ref") and self._needs_ref(vals=vals):
+            vals["ref"] = self._get_next_ref()
+        return super(ResPartner, self).create(vals)
```

Comment:
@Nikul-OSI The model odoo.addons.base_partner_sequence.models.partner is not overriding the create method in batch 
```suggestion
    @api.model_create_multi
    def create(self, vals_list):
        for vals in vals_list:
            if not vals.get("ref") and self._needs_ref(vals=vals):
                vals["ref"] = self._get_next_ref()
        return super(ResPartner, self).create(vals_list)
```